kyo-core/shared/src/main/scala/kyo/Resource.scala

```diff
@@ -24,14 +40,43 @@ object Resource:
             }
         }
 
+    /** Acquires a resource and provides a release function.
+      *
+      * @param acquire
+      *   The effect to acquire the resource.
+      * @param release
+      *   The function to release the acquired resource.
+      * @param frame
+      *   The implicit Frame for context.
+      * @return
+      *   The acquired resource wrapped in Resource, IO, and S effects.
+      */
     def acquireRelease[A, S](acquire: A < S)(release: A => Unit < Async)(using Frame): A < (Resource & IO & S) =
         acquire.map { resource =>
             ensure(release(resource)).andThen(resource)
         }
 
+    /** Acquires a Closeable resource.
+      *
+      * @param resource
+      *   The effect to acquire the Closeable resource.
+      * @param frame
+      *   The implicit Frame for context.
+      * @return
+      *   The acquired Closeable resource wrapped in Resource, IO, and S effects.
+      */
     def acquire[A <: Closeable, S](resource: A < S)(using Frame): A < (Resource & IO & S) =
         acquireRelease(resource)(r => IO(r.close()))
```

Comment:
I think we need to ensure that `r` is not null: https://github.com/zio/zio/pull/9008/files